### PR TITLE
1487600: Fix registration success detection

### PR DESCRIPTION
**Pull request by kahowell**

### CODE REVIEW

**Comment by kahowell** [comment]
> When you type wrong proxy servername/username/password, then the registration does not fail (candlepin has to be available without proxy) and wrong values are saved to rhsm.conf. May be it is problem of D-Bus object implementation.

We have a bit of an implementation detail that leaked across the API... All of the interfaces (CLI, GUI, and now cockpit), handle defaulting of proxy port individually. (`git grep DEFAULT_PROXY_PORT` illustrates this). For consistency I went ahead and just handled it in the cockpit plugin code as well...

**Comment by jirihnidek** [comment]
I also noticed this: When you try to register using wrong proxy settings (wrong proxy server), then error message is raised as expected. When you try to register second time with some other wrong proxy settings, then subman cockpit plugin is frozen for several minutes. The problem is probably in rhsm-service in `rhsm.connection` module.

**Comment by kahowell** [comment]
> When you try to register using wrong proxy settings (wrong proxy server), then error message is raised as expected. When you try to register second time with some other wrong proxy settings, then subman cockpit plugin is frozen for several minutes.

Have not seen this with latest changes:
- default everything in one place, before parsing user input.
- handle proxy options in the defaults.
- made a function to shorten creating D-Bus strings; this made the code a lot shorter.

File: cockpit/src/subscriptions-client.js [hunk 1]
```diff
@@ -121,9 +122,27 @@ function getSubscriptionDetails() {
...
+    // values. Otherwise old and obsolete values from rhsm.conf could be used.
+    const connection_options = {
+        host: s(RHSM_DEFAULTS.hostname),
+        port: s(RHSM_DEFAULTS.port),
+        proxy_hostname: s(''),
+        proxy_port: s(''),
+        proxy_user: s(''),
+        proxy_password: s(''),
+        handler: s(RHSM_DEFAULTS.prefix),
+    };
```
Comment (jirihnidek) [inline cockpit/src/subscriptions-client.js]: It simplifies code, but it breaks resetting of config values. The `if` statement on the line 299 is always true.

File: cockpit/src/subscriptions-client.js [hunk 2]
```diff
@@ -121,9 +122,27 @@ function getSubscriptionDetails() {
     });
 }
 
+/* convenience function for specifying d-bus strings */
+function s(value) {
```
Comment (jirihnidek) [inline cockpit/src/subscriptions-client.js]: I like this refactoring 👍. Can we rename it to something like `dbus_str()`?

File: cockpit/src/subscriptions-client.js (hunk 1 above)
Comment (kahowell) [reply cockpit/src/subscriptions-client.js]: > It simplifies code, but it breaks resetting of config values. The if statement on the line 299 is always true.

Hmmm. I'm thinking this is a not a real issue in terms of functionality, but I'll make a commit to remove dead code/simplify.

I'm leaving explanation in the change.

File: cockpit/src/subscriptions-client.js (hunk 2 above)
Comment (kahowell) [reply cockpit/src/subscriptions-client.js]: Done

**Comment by kahowell** [comment]
b8741bb6..24ea7e8b: squashed

@jirihnidek FYI

**Pull request merged**